ldaptive/src/main/java/org/ccci/idm/user/ldaptive/dao/io/GroupValueTranscoder.java

```diff
@@ -1,45 +1,79 @@
 package org.ccci.idm.user.ldaptive.dao.io;
 
-import com.google.common.base.Strings;
-import com.google.common.collect.ImmutableList;
-import com.google.common.collect.Lists;
+import com.google.common.annotations.VisibleForTesting;
 import org.ccci.idm.user.Dn;
 import org.ccci.idm.user.Group;
-import org.ldaptive.DnParser;
-import org.ldaptive.LdapAttribute;
+import org.ccci.idm.user.ldaptive.dao.util.DnUtils;
 import org.ldaptive.io.AbstractStringValueTranscoder;
 
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
-import java.util.List;
 
 public class GroupValueTranscoder extends AbstractStringValueTranscoder<Group> {
-    private static final String delimiter = ",";
-    private static final String valueDelimiter = "=";
-
-    private String baseDn = "";
+    @Nonnull
+    private Dn baseDn = Dn.ROOT;
 
+    /**
+     * @deprecated Since 0.3.0, kept for old spring config compatibility.
+     */
+    @Nonnull
+    @Deprecated
     public String getBaseDn() {
-        return this.baseDn;
+        return getBaseDnString();
+    }
+
+    @Nonnull
+    @VisibleForTesting
+    String getBaseDnString() {
```

Comment:
This has to do with Spring's rules for setting properties and keeping some backwards compatibility for now. Spring tries to intelligently guess the type of the property being set, it checks the getter first, followed by the property type and/or the setter type.

We are currently overloading the setter to handle both `Dn` and `String`s which thoroughly confuses Spring unless you have a getter return the type you want to use from Spring config.

Long term I want to keep the support for setting both, but needed to separate the setter names to keep spring happy (long term `setBaseDn(Dn)` and `setBaseDnString(String)` will be the 2 setters).

The getter response types should match the setter types, so `getBaseDn()` should return a `Dn`, but to keep spring happy with backwards compatibility I needed to keep it returning a String for now.

Although, now that we already broke some backwards compatibility by removing `getPath()`, it probably wouldn't hurt to drop this backwards compatibility support now as well and fix our XML config where ever we create a `GroupValueTranscoder`.